perllib/FixMyStreet/Cobrand/FixMyStreet.pm

```diff
@@ -62,5 +65,96 @@ sub extra_contact_validation {
     return %errors;
 }
 
-1;
+=head2 only_council_user
+
+This is for council-specific dashboard pages, which can only be seen by
+superusers and logged-in users with an email domain matching a body name.
+
+=cut
+
+sub only_council_user {
+    my $self = shift;
+    my $c = $self->{c};
+
+    unless ( $c->user_exists ) {
+        $c->res->redirect('/about/council-dashboard');
+        $c->detach;
+    }
+
+    return if $c->user->is_superuser;
+
+    if (my $body = _user_to_body($c)) {
+        # Matching URL and user's email body
+        return if $body->id eq $c->stash->{body}->id;
+
+        # Matched /a/ body, redirect to its summary page
+        $c->stash->{body} = $body;
+        $c->stash->{wards} = [ { name => 'summary' } ];
+        $c->detach('/reports/redirect_body');
+    }
+
+    $c->res->redirect('/about/council-dashboard');
+}
```

Comment:
Should this not also check if they are flagged as belonging to the council in the database?

Reply:
No, as I understand it we're letting anyone with a council email address see these pages, whether they're a staff user or not.

Reply:
Sorry, wasn't very clear. I meant as an extra check in the same way we check if they are a superuser. Just on the off chance that they are a staff user but don't have a matching email for whatever reason.

Reply:
Aha, right, yep, will do.